### PR TITLE
feat(): Only send emails when confirmations are on

### PR DESCRIPTION
**Pull request by TylorS**
WIP don't merge yet
- [x] Don't send engagement email if confirmations are off
- [x] Send emails to all accepted applicants when confirmations are turned on
- [x] Send emails to new applicants after confirmations are on

### CODE REVIEW

File: src/tasks/Opps.js [hunk 1]
```diff
@@ -27,7 +29,31 @@ const remove = (key, uid, {Profiles, Opps, Projects}) =>
...
+    .then(engagements => engagements.filter(e => e.isAccepted))
+}
+
+function checkAndSendAcceptanceEmail(key, {confirmationsOn}, uid, opp, Engagements, Profiles, Opps, Projects) { // eslint-disable-line
+  console.log(opp)
+  if (confirmationsOn && !opp.hasOwnProperty(confirmationsOn)) {
+    process.nextTick(() => {
+      getAcceptedApplicants(Engagements, opp.key)
+        .then(engagements => {
+          engagements.forEeach(a => {
```
Comment (jemmyw) [inline src/tasks/Opps.js]: forEeach -> forEach

**Comment by jemmyw** [comment]
When sending the email to applications should we be marking that we have sent it to them somewhere in the database so that we don't send it to them again if the organizer turns off that setting and turns it back on again?

File: src/tasks/Opps.js [hunk 2]
```diff
@@ -36,9 +62,10 @@ const update = ({key, values}, uid, {Profiles, Opps, Projects}) =>
     Projects.get(opp.projectKey).then(project => [user, opp, project])
   )
   .then(([user, opp, project]) =>
-    (isUser(user,opp.ownerProfileKey) ||
+    checkAndSendAcceptanceEmail(key, values, uid, opp, Engagements, Profiles, Opps, Projects) || // eslint-disable-line max-len
```
Comment (jemmyw) [inline src/tasks/Opps.js]: doesn't putting this here cause it to queue sending the email before we're sure the opp will be updated?

Comment (TylorS) [inline src/tasks/Opps.js]: Yes, you're correct. I'll make sure to fix that tonight

**Comment by TylorS** [comment]
> When sending the email to applications should we be marking that we have sent it to them somewhere

Probably. Is it wise to even allow turning off confirmations? E.g. those people will still get an email that they are on, but would (not true yet) be unable to follow the links to confirm.

**Comment by jemmyw** [comment]
Yes that seems sensible, we need to update the frontend to prevent turning it off and put explanation there

**Comment by TylorS** [comment]
@jemmyw This now works appropriately :+1:

**Comment by jemmyw** [comment]
🚢 looks good

**Pull request merged**